tools/create_db.py

```diff
@@ -448,7 +454,21 @@ def write_batch(self, batch):
         keys = self.get_keys(len(batch))
         lmdb_txn = self.db.begin(write=True)
         for i, datum in enumerate(batch):
-            lmdb_txn.put('%08d_%d' % (keys[i], datum.label), datum.SerializeToString())
+            try:
+                key = '%08d_%d' % (keys[i], datum.label)
+                lmdb_txn.put(key, datum.SerializeToString())
+            except lmdb.MapFullError:
+                lmdb_txn.abort()
+
+                # double the map_size
+                curr_limit = self.db.info()['map_size']
+                new_limit = curr_limit*2
+                logger.debug('Doubling LMDB map size to %sMB ...' % (new_limit>>20,))
+                self.db.set_mapsize(curr_limit*2) # double it
```

Comment:
since this is a new API in py-lmdb, is it worth adding a check that version is >0.86? I notice the version has not been updated yet in https://github.com/dw/py-lmdb/blob/master/lmdb/__init__.py but I guess this will come soon.